tests/test_timing.py

```diff
@@ -0,0 +1,305 @@
+from dmutils import timing
+
+from collections import OrderedDict
+from contextlib import contextmanager
+from functools import lru_cache
+from itertools import product, zip_longest
+import json
+import logging
+import mock
+from numbers import Number
+import re
+import sys
+from time import sleep
+
+from flask import request
+import pytest
+
+
+class malleable_ANY:
+    def __init__(self, condition):
+        self._condition = condition
+
+    def __eq__(self, other):
+        return self._condition(other)
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}({self._condition})"
+
+    def __hash__(self):
+        return None
+
+
+class ANY_superset_of(malleable_ANY):
+    def __init__(self, subset_dict):
+        self._subset_dict = subset_dict
+        super().__init__(lambda other: self._subset_dict == {k:v for k, v in other.items() if k in self._subset_dict})
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}({self._subset_dict})"
+
+
+class ANY_string_matching(malleable_ANY):
+    _cached_re_compile = staticmethod(lru_cache(maxsize=32)(re.compile))
+
+    def __init__(self, *args, **kwargs):
+        self._regex = self._cached_re_compile(*args, **kwargs)
+        super().__init__(lambda other: self._regex.match(other))
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}({self._regex})"
+
+
+# python 3.7 has an in-built version of this, but for now...
+@contextmanager
+def null_context_manager():
+    yield
+
+
+class SentinelError(Exception):
+    """An exception that can't be mistaken for a genuine problem"""
+    pass
+
+
+def _duration_real_gt_048(log_context):
+    return log_context["duration_real"] > 0.095
```

Comment:
Method name here doesn't match up with the actual value.

Reply:
Well that would be *too* obvious wouldn't it?